test/connectionUI.test.ts

```diff
@@ -60,44 +64,62 @@ suite('Connection UI tests', () => {
 		outputChannel.verify(c => c.show(true), TypeMoq.Times.once());
 	});
 
-	test('showConnections with recent and new connection', () => {
+	test('showConnections with recent and new connection', async () => {
 		let item: IConnectionCredentialsQuickPickItem = {
 			connectionCreds: undefined,
 			quickPickItemType: CredentialsQuickPickItemType.NewConnection,
 			label: undefined
 		};
+		quickPickMock.object.items = [item];
```

Comment:
Do you actually need to set this? The prompt just gets the selected item via the callback, the items property is only used to set the items (which this test doesn't currently care about).

But also you should consider capturing the values passed to the items setter and then use those items in the onDidChangeSelect callback. Because it's the product code that actually sets the properties - by overriding that here you're skipping part of the actual functionality. 

(same for others below)

Reply:
Removed this since as you stated we send it via the emitter/callback.